lib/aws-manager.js

```diff
@@ -81,142 +82,164 @@ class AwsManager {
+            {
+              Name: 'instance-state-name',
+              Values: ['running', 'pending'],
+            },
+          ],
+        }).promise(),
+        // Living spot requests
+        obj.describeSpotInstanceRequests({
+          Filters: [
+            {
+              Name: 'launch.key-name',
+              Values: [this.keyPrefix + '*'],
+            }, {
+              Name: 'state',
+              Values: ['open'],
+            },
+          ],
+        }).promise(),
+        // Dead instances
+        obj.describeInstances({
+          Filters: [
+            {
+              Name: 'key-name',
+              Values: [this.keyPrefix + '*'],
+            },
+            {
+              Name: 'instance-state-name',
+              Values: ['shutting-down', 'terminated', 'stopping'],
+            },
+          ],
+        }).promise(),
+        // Dead spot requests
+        obj.describeSpotInstaceRequests({
+          Filters: [
+            {
+              Name: 'launch.key-name',
+              Values: [this.keyPrefix + '*'],
+            }, {
+              Name: 'state',
+              Values: ['cancelled', 'failed', 'closed', 'active'],
+            },
+          ],
+        }).promise(),
+        // Available availability zones
+        obj.describeAvailabilityZones({
+          Filters: [
+            {
+              Name: 'state',
+              Values: ['available'],
+            },
+          ],
+        }).promise(),
+        // Raw pricing data
+        obj.describeSpotPriceHistory({
+          StartTime: pricingStartDate,
+          Filters: [
+            {
+              Name: 'product-description',
+              Values: ['Linux/UNIX'],
+            },
+          ],
+        }).promise(),
+      ]);
 
-      deadInstances = await this.ec2.describeInstances({
-        Filters: [
-          {
-            Name: 'key-name',
-            Values: [this.keyPrefix + '*'],
-          },
-          {
-            Name: 'instance-state-name',
-            Values: ['shutting-down', 'terminated', 'stopping'],
-          },
-        ],
-      });
-      debug('fetched dead instance state');
+      // Now let's classify them
 
-      let rawDeadSR = await this.ec2.describeSpotInstanceRequests({
-        Filters: [
-          {
-            Name: 'launch.key-name',
-            Values: [this.keyPrefix + '*'],
-          }, {
-            Name: 'state',
-            Values: ['cancelled', 'failed', 'closed', 'active'],
-          },
-        ],
-      });
-      debug('fetched dead spot request state');
+      for (let reservation of response[0].data.Reservations) {
+        for (let instance of reservation.Instances) {
+          let workerType = this.parseKeyPairName(instance.KeyName).workerType;
+          // Maybe use objFilter here
+          let filtered = instance;
+          filtered.Region = region;
+          filtered.WorkerType = workerType;
+          apiState.instances.push(filtered);
+        }
+      };
```

Comment:
**Sorry, I'm wrong...**

You can't do this because the call for each region, adds to this list in parallel..